analyzers/src/SonarAnalyzer.CSharp/Helpers/MutedSyntaxWalker.cs

```diff
@@ -0,0 +1,130 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.ShimLayer.CSharp;
+
+namespace SonarAnalyzer.Helpers
+{
+    /// <summary>
+    /// This class find syntax cases that are not properly supported by current CFG/SE/LVA and we mute all issues related to these scenarios.
+    /// </summary>
+    internal class MutedSyntaxWalker : CSharpSyntaxWalker
+    {
+        // All kinds that SonarAnalysisContextExtensions.RegisterExplodedGraphBasedAnalysis registers for
+        private static readonly SyntaxKind[] RootKinds = new[]
+        {
+                SyntaxKind.ConstructorDeclaration,
+                SyntaxKind.DestructorDeclaration,
+                SyntaxKind.ConversionOperatorDeclaration,
+                SyntaxKind.OperatorDeclaration,
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.PropertyDeclaration,
+                SyntaxKind.GetAccessorDeclaration,
+                SyntaxKind.SetAccessorDeclaration,
+                SyntaxKind.AddAccessorDeclaration,
+                SyntaxKind.RemoveAccessorDeclaration,
+                SyntaxKind.AnonymousMethodExpression,
+                SyntaxKind.SimpleLambdaExpression,
+                SyntaxKind.ParenthesizedLambdaExpression
+            };
+
+        private readonly SemanticModel semanticModel;
+        private readonly SyntaxNode root;
+        private readonly ISymbol[] symbols;
+        private bool isMuted;
+        private bool isInTryOrCatch;
+        private bool isOutsideTryCatch;
+
+        public MutedSyntaxWalker(SemanticModel semanticModel, SyntaxNode node)
+            : this(semanticModel, node, node.DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().Select(x => semanticModel.GetSymbolInfo(x).Symbol).WhereNotNull().ToArray()) { }
+
+        public MutedSyntaxWalker(SemanticModel semanticModel, SyntaxNode node, params ISymbol[] symbols)
+        {
+            this.semanticModel = semanticModel;
+            this.symbols = symbols;
+            root = node.Ancestors().FirstOrDefault(x => x.IsAnyKind(RootKinds));
+        }
+
+        public bool IsMuted()
+        {
+            if (symbols.Any() && root != null)
+            {
+                Visit(root);
+            }
+            return isMuted || (isInTryOrCatch && isOutsideTryCatch);
```

Comment:
It would be nice to add a comment explaining why we look for both in and outside of trycatch.

Reply:
It's quite clear from the variable names IMO